File: src/main/java/com/race/codeDash/dto/RaceDataDto.java
package com.race.codeDash.dto;

import com.race.codeDash.infrastructure.TupleLineCode;

import java.util.LinkedList;

public record RaceDataDto(
		String creatorName,
		String difficulty,
		String programmingLanguage,
		Integer time,
		TupleLineCode<LinkedList<Integer>, LinkedList<CodeDto>> lineCode
) {
}

File: codeDashPhp/model/dto/RaceDataDto.php
<?php

namespace Model\Dto;


class RaceDataDto {
    
    public  $creatorName;
    public  $difficulty;
    public  $programmingLanguage;
    public  $time;
    public $description;
    public  $lineCode;

    public $raceStats;

    public $createdAt;

    public $codeId;

    public function __construct(
        $creatorName,
        $difficulty,
        $programmingLanguage,
        $time,
        $lineCode,
        $raceStats,
        $createdAt,
        $codeId,
        $description = null,
    ) {
        $this->creatorName = $creatorName;
        $this->difficulty = $difficulty;
        $this->programmingLanguage = $programmingLanguage;
        $this->time = $time;
        $this->lineCode = $lineCode;
        $this->description = $description;
        $this->raceStats = $raceStats;
        $this->createdAt = $createdAt;
        $this->codeId = $codeId;
    }
}